rust/private/rustc.bzl

```diff
@@ -1748,7 +1748,7 @@ def _portable_link_flags(lib, use_pic, ambiguous_libs, get_lib_name, for_windows
             return [] if for_darwin else ["-lstatic=%s" % get_lib_name(artifact)]
         return [
             "-lstatic=%s" % get_lib_name(artifact),
-            "-Clink-arg=-l%s" % (get_lib_name(artifact) if not for_windows else artifact.basename),
+            "-Clink-arg={}{}".format("-l" if not flavor_msvc else "", get_lib_name(artifact)) if not for_windows else "-Clink-arg={}".format(artifact.basename),
```

Comment:
Style nit: Could you make a list that you conditionally append the right thing into? This conditional seems kinda long.

Reply:
Will do.  Is the `msvc_flavor` change okay or would it be better to curry in a reference to `toolchain` and get rid of `for_windows` and `for_darwin` in favor of `target_triple` checks?